### ISSUE TITLE
Trouble with CapsLock

### ISSUE THREAD

**Issue by Syphdias** [mihi314/neolight#1]
Hi, I'd love to use mainly the 3rd layer but I am having trouble to get any change in behavior. Especially the Capslock acts like normal instead of becoming a modifier. I installed neolight through AUR and tried to select it from the ibus input picker at first but it didn't show up. I tried to do it manually but with not luck. Do you have any ideas what I am doing wrong? Maybe the xkb-options are wrong?

```console
❯ localectl
   System Locale: LANG=en_US.UTF-8
                  LC_NUMERIC=de_DE.UTF-8
                  LC_TIME=en_GB.UTF-8
                  LC_MONETARY=de_DE.UTF-8
                  LC_PAPER=de_DE.UTF-8
                  LC_NAME=de_DE.UTF-8
                  LC_ADDRESS=de_DE.UTF-8
                  LC_TELEPHONE=de_DE.UTF-8
                  LC_MEASUREMENT=de_DE.UTF-8
                  LC_IDENTIFICATION=de_DE.UTF-8
       VC Keymap: de-latin1-nodeadkeys
      X11 Layout: de
       X11 Model: pc105
     X11 Variant: nodeadkeys
❯ ibus engine
xkb:de:nodeadkeys:deu
❯ dconf dump /org/gnome/desktop/input-sources/
[/]
show-all-sources=true
sources=[('xkb', 'de+nodeadkeys')]
xkb-options=['neolight:escape_keys']
❯ dconf dump /desktop/ibus/general/
[/]
engines-order=['xkb:de:nodeadkeys:deu', 'xkb:de:neo:deu', 'xkb:us::eng']
preload-engines=['xkb:de:nodeadkeys:deu']
use-system-keyboard-layout=false
version='1.5.26'
```

i3 on Manjaro

**Comment by mihi314** [comment]
Hi!

Yea, so unfortunately every desktop environment / input method framework has its own way of handling xkb options and layouts.

This is not a permanent solution, but you can try to set the layout with xkb directly, to check whether it is at least installed:
```setxkbmap -layout neolight -variant de_escape_keys```
or
```setxkbmap -option neolight:escape_keys```
And then try it out and check whether it shows up in `setxkbmap -query`. (This all assumes you've already reloaded the xkb config after installing neolight, which is most reliably done by logging out and back in again or restarting).


Also are you using gnome? If not, then setting the `/org/gnome/desktop/input-sources/xkb-options` dconf entry won't have an effect. (But the entry itself looks correct to me.)

For making it show up in the ibus picker (`ibus-setup`?) have a look here: https://askubuntu.com/a/1242981. You'll need to add an entry like this (and then restart the `ibus-daemon`):

```
<engine>
    <name>xkb:de:neolight:deu</name>
    <language>de</language>
    <license></license>
    <author></author>
    <layout>neolight</layout>
    <layout_variant>de_escape_keys</layout_variant>
    <longname>German (Neolight)</longname>
    <description>German (Neolight)</description>
    <icon>ibus-keyboard</icon>
    <rank>1</rank>
</engine>
```
If this works for you and I find the time I'll maybe add it to the AUR package / install script.

**Comment by Syphdias** [comment]
I admit to trying the GNOME setting in a desperate attempt that `ibus-setup` would pick up on it as well for compatibility reasons – it does not.

Kudos for figuring out my problem from the little info I gave – sorry for that it was late.

I think I prefer the no escape version for now to get `+` and `*`. Might change in the future though.
The addition to `/usr/share/ibus/component/simple.xml` worked, but I named them a bit differently:
`xkb:de:neolight:deu` and `xkb:de:neolight_escape_keys:deu`

```
		<engine>
		        <name>xkb:de:neolight:deu</name>
		        <language>de</language>
		        <license></license>
		        <author></author>
		        <layout>neolight</layout>
		        <longname>German (Neolight)</longname>
		        <description>German (Neolight)</description>
		        <icon>ibus-keyboard</icon>
		        <rank>1</rank>
		</engine>
		<engine>
		        <name>xkb:de:neolight_escape_keys:deu</name>
		        <language>de</language>
		        <license></license>
		        <author></author>
		        <layout>neolight</layout>
		        <layout_variant>de_escape_keys</layout_variant>
		        <longname>German (Neolight with escape keys)</longname>
		        <description>German (Neolight with escape keys)</description>
		        <icon>ibus-keyboard</icon>
		        <rank>1</rank>
		</engine>
```

I am not sure about the proper naming here.

PS: Depending on who has the time first, I can also add the modifications. It might take a bit more modifications since the current register function assumes it only needs to change stuff in one directory.

**Comment by mihi314** [comment]
Nice that it worked! You provided quite a bit of information, the config dumps were very helpful :D

> I am not sure about the proper naming here.

Elsewhere I've named the escape key version ["German (Neolight, additional escape keys)"](https://github.com/mihi314/neolight/blob/f6e683e8e5ea44753e0185965246b8c362330133/linux/register-neolight.sh#L30), so can just go with that here as well.

Yea, the register script/function will need to be changed a bit. Feel free to tackle that if you want. I'd add a check for whether `/usr/share/ibus/component/simple.xml` is present and then handle it similar to the other cases.

**Comment by Syphdias** [comment]
I just realized the layout gets lost if I reload my wm after doing changes to the config (`i3-msg reload`). I assume this has always happened but since the `localectl` is was to the same layout (a sane fallback, just in case). This is clearly an issue with the `ibus-daemon` not reloading/restarting and not of neolight.

For documentation purposes – I replaced `exec` with `exec_always` for the daemon, so it replaced the old one, thus "restarting" on i3 reload:
```
exec_always --no-startup-id ibus-daemon -drx
```
